File: liqp-core/src/main/java/liqp/tags/Capture.java
package liqp.tags;

import liqp.tag.LTag;
import liqp.node.LNode;
import liqp.context.LContext;

public class Capture extends LTag {

    /*
     * Block tag that captures text into a variable
     */
    @Override
    public Object render(LContext context, LNode... nodes) {

        String id = super.asString(nodes[0].render(context));

        LNode block = nodes[1];

        // Capture causes variable to be saved "globally"
        context.set(id, block.render(context));

        return null;
    }
}

File: liqp-core/src/main/kotlin/liqp/tags/Capture.kt
package liqp.tags

import liqp.ControlResult
import liqp.context.LContext
import liqp.node.LNode
import liqp.tag.LTag

class Capture : LTag() {

  /**
   * Block tag that captures text into a variable
   */
  override fun render(context: LContext, vararg nodes: LNode): Any? {

    val id:String = nodes[0].execute(context)
    val block = nodes[1]

    // Capture causes variable to be saved "globally"
    context[id] = block.render(context)

    return ControlResult.NO_CONTENT
  }
}
